pkg/tbtcpg/chain_test.go

```diff
@@ -687,13 +924,169 @@ func (lc *LocalChain) GetWallet(walletPublicKeyHash [20]byte) (
 	*tbtc.WalletChainData,
 	error,
 ) {
+	lc.mutex.Lock()
+	defer lc.mutex.Unlock()
+
+	data, ok := lc.walletChainData[walletPublicKeyHash]
+	if !ok {
+		fmt.Println("Not found")
+		return nil, fmt.Errorf("wallet chain data not found")
+	}
+
+	return data, nil
+}
+
+func (lc *LocalChain) SetWallet(
+	walletPublicKeyHash [20]byte,
+	data *tbtc.WalletChainData,
+) {
+	lc.mutex.Lock()
+	defer lc.mutex.Unlock()
+
+	lc.walletChainData[walletPublicKeyHash] = data
+}
+
+func (lc *LocalChain) GetWalletParameters() (
+	creationPeriod uint32,
+	creationMinBtcBalance uint64,
+	creationMaxBtcBalance uint64,
+	closureMinBtcBalance uint64,
+	maxAge uint32,
+	maxBtcTransfer uint64,
+	closingPeriod uint32,
+	err error,
+) {
+	lc.mutex.Lock()
+	defer lc.mutex.Unlock()
+
+	return lc.walletParameters.creationPeriod,
+		lc.walletParameters.creationMinBtcBalance,
+		lc.walletParameters.creationMaxBtcBalance,
+		lc.walletParameters.closureMinBtcBalance,
+		lc.walletParameters.maxAge,
+		lc.walletParameters.maxBtcTransfer,
+		lc.walletParameters.closingPeriod,
+		nil
+}
+
+func (lc *LocalChain) SetWalletParameters(
+	creationPeriod uint32,
+	creationMinBtcBalance uint64,
+	creationMaxBtcBalance uint64,
+	closureMinBtcBalance uint64,
+	maxAge uint32,
+	maxBtcTransfer uint64,
+	closingPeriod uint32,
+) {
+	lc.mutex.Lock()
+	defer lc.mutex.Unlock()
+
+	lc.walletParameters = walletParameters{
+		creationPeriod:        creationPeriod,
+		creationMinBtcBalance: creationMinBtcBalance,
+		creationMaxBtcBalance: creationMaxBtcBalance,
+		closureMinBtcBalance:  closureMinBtcBalance,
+		maxAge:                maxAge,
+		maxBtcTransfer:        maxBtcTransfer,
+		closingPeriod:         closingPeriod,
+	}
+}
+
+func (lc *LocalChain) GetLiveWalletsCount() (uint32, error) {
 	panic("unsupported")
 }
 
 func (lc *LocalChain) ComputeMainUtxoHash(mainUtxo *bitcoin.UnspentTransactionOutput) [32]byte {
 	panic("unsupported")
 }
 
+func (lc *LocalChain) ComputeMovingFundsCommitmentHash(targetWallets [][20]byte) [32]byte {
+	packedWallets := []byte{}
+
+	for _, wallet := range targetWallets {
+		packedWallets = append(packedWallets, wallet[:]...)
+		// Each wallet hash must be padded with 12 zero bytes following the
+		// actual hash.
+		packedWallets = append(packedWallets, make([]byte, 12)...)
```

Comment:
Not relevant for actual code but we may want to remove it to avoid confusion if it turns out that this padding is not necessary (see previous comment).

Reply:
The padding seems to be necessary, see https://github.com/keep-network/keep-core/pull/3748#discussion_r1482681860.